samples/file-encryptor/enclave/keys.cpp

```diff
@@ -290,9 +290,17 @@ int ecall_dispatcher::prepare_encryption_header(
     {
         TRACE_ENCLAVE("password_key");
         for (unsigned int i = 0; i < ENCRYPTION_KEY_SIZE_IN_BYTES; i++)
+        {
+            if (password_key[i] > UINT32_MAX)
```

Comment:
`password_key[i]` is a 1 byte unsigned char. It isn't possible to be larger than UINT32_MAX.

I am guessing you're meaning to check if all of password key (which is an array of bytes) can fit into a uint32?

Even then, isn't this check redundant? ENCRYPTION_KEY_SIZE_IN_BYTES is 32, so by definition, it can fit into a uint32

Reply:
ENCRYPTION_KEY_SIZE_IN_BYTES  = 32 is the size of the buffer. Since we are casting password+key[i] to unsigned it, they check looks right to me. @jhand2 could you please elaborate here? Thanks.

Reply:
My mistake, the first thing I said was right but the part about it always fitting into a uint32 was wrong.

Still I’m a bit confused about what we’re checking here. As is this just checks that each byte in password_key fits into a uint32 which will always be true.

Reply:
I agree with Jordan, this change doesn't make sense to me either